### DISCUSSION TITLE
A matching Triton is not available, some optimizations will not be enabled. Error caught was: No module named 'triton'

### DISCUSSION THREAD

**Discussion by kou201** [kohya-ss/sd-scripts#319] [Q&A]
LoRA type changed...
Folder 10_onineko-2: 95 images found
Folder 10_onineko-2: 950 steps
max_train_steps = 4750
stop_text_encoder_training = 0
lr_warmup_steps = 238
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
accelerate launch --num_cpu_threads_per_process=16 "train_network.py" --enable_bucket --pretrained_model_name_or_path="I:/Koikatsu/NoMust/novelai/novelai-webui-10-9update/models/Stable-diffusion/model-latest.ckpt" --train_data_dir="I:\Koikatsu\NoMust\novelai\训练文件整合\lora\onineko-2" --resolution=512,512 --output_dir="I:/Koikatsu/NoMust/novelai/novelai-webui-10-9update/outputs/Lora-Outpu" --logging_dir="I:\Koikatsu\NoMust\novelai\novelai-webui-10-9update\outputs\Lora-Output\lora log" --network_alpha="64" --training_comment="painting style" --save_model_as=safetensors --network_module=lycoris.kohya --network_args "conv_dim=11" "conv_alpha=1" "algo=loha" --text_encoder_lr=7e-5 --unet_lr=6e-5 --network_dim=128 --output_name="Onineko-loha" --lr_scheduler_num_cycles="10" --learning_rate="6e-5" --lr_scheduler="cosine_with_restarts" --lr_warmup_steps="238" --train_batch_size="2" --max_train_steps="4750" --save_every_n_epochs="2" --mixed_precision="fp16" --save_precision="fp16" --seed="31337" --cache_latents --optimizer_type="Lion" --max_token_length=225 --clip_skip=2 --bucket_reso_steps=64 --save_state --xformers --persistent_data_loader_workers --bucket_no_upscale --noise_offset=0.1  --sample_sampler=euler_a --sample_prompts="I:/Koikatsu/NoMust/novelai/novelai-webui-10-9update/outputs/Lora-Outpu\sample\prompt.txt" --sample_every_n_epochs="1"
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
prepare tokenizer
update token length: 225
Use DreamBooth method.
prepare images.
found directory I:\Koikatsu\NoMust\novelai\训练文件整合\lora\onineko-2\10_onineko-2 contains 190 image files
1900 train images with repeating.
0 reg images.
no regularization images / 正則化画像が見つかりませんでした
[Dataset 0]
  batch_size: 2
  resolution: (512, 512)
  enable_bucket: True
  min_bucket_reso: 256
  max_bucket_reso: 1024
  bucket_reso_steps: 64
  bucket_no_upscale: True

  [Subset 0 of Dataset 0]
    image_dir: "I:\Koikatsu\NoMust\novelai\训练文件整合\lora\onineko-2\10_onineko-2"
    image_count: 190
    num_repeats: 10
    shuffle_caption: False
    keep_tokens: 0
    caption_dropout_rate: 0.0
    caption_dropout_every_n_epoches: 0
    caption_tag_dropout_rate: 0.0
    color_aug: False
    flip_aug: False
    face_crop_aug_range: None
    random_crop: False
    is_reg: False
    class_tokens: onineko-2
    caption_extension: .caption


[Dataset 0]
loading image sizes.
100%|████████████████████████████████████████████████████████████████████████████████| 95/95 [00:00<00:00, 3873.76it/s]
make buckets
min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is set, because bucket reso is defined by image size automatically / bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計算されるため、min_bucket_resoとmax_bucket_resoは無視されます
number of images (including repeats) / 各bucketの画像枚数（繰り返し回数を含む）
bucket 0: resolution (512, 512), count: 950
mean ar error (without repeats): 1.8237427962950937e-05
prepare accelerator
Using accelerator 0.15.0 or above.
load StableDiffusion checkpoint
loading u-net: <All keys matched successfully>
loading vae: <All keys matched successfully>
loading text encoder: <All keys matched successfully>
Replace CrossAttention.forward to use xformers
[Dataset 0]
caching latents.
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 95/95 [00:12<00:00,  7.66it/s]
import network module: lycoris.kohya
Using rank adaptation algo: loha
====================WARNING====================
I:\Koikatsu\NoMust\novelai\lora\kohya_ss\train_network.py:162: UserWarning: You are not supposed to use dim>64 (64*64 = 4096, it already has enough rank)in Hadamard Product representation!
Please consider use lower dim or disable this warning with --network_args no_dim_warn=True
If you just want to use high dim loha, please consider use lower lr.
  network = network_module.create_network(1.0, args.network_dim, args.network_alpha, vae, text_encoder, unet, **net_kwargs)
====================WARNING====================
Apply different lora dim for conv layer
Conv Dim: 11, Linear Dim: 128
Apply different alpha value for conv layer
Conv alpha: 1.0, Linear alpha: 64.0
Use Dropout value: 0.0
Create LyCORIS Module
create LyCORIS for Text Encoder: 72 modules.
Create LyCORIS Module
create LyCORIS for U-Net: 278 modules.
enable LyCORIS for text encoder
enable LyCORIS for U-Net
prepare optimizer, data loader etc.
use Lion optimizer | {}
running training / 学習開始
  num train images * repeats / 学習画像の数×繰り返し回数: 1900
  num reg images / 正則化画像の数: 0
  num batches per epoch / 1epochのバッチ数: 475
  num epochs / epoch数: 10
  batch size per device / バッチサイズ: 2
  gradient accumulation steps / 勾配を合計するステップ数 = 1
  total optimization steps / 学習ステップ数: 4750
steps:   0%|                                                                                                                                                                                                                                                  | 0/4750 [00:00<?, ?it/s]epoch 1/10
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'


what's going wrong?
